src/sfdc48/salesforce_bulk_endpoint.bal

```diff
@@ -0,0 +1,369 @@
+//
+// Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import ballerina/http;
+import ballerina/oauth2;
+import ballerina/io;
+
+# Salesforce Client object.
+# + httpClient - OAuth2 client endpoint
+# + salesforceConfiguration - Salesforce Connector configuration
+public type BulkClient client object {
+    http:Client httpClient;
+    SalesforceConfiguration salesforceConfiguration;
+
+    # Salesforce Connector endpoint initialization function.
+    # + salesforceConfig - Salesforce Connector configuration
+    public function __init(SalesforceConfiguration salesforceConfig) {
+        self.salesforceConfiguration = salesforceConfig;
+        // Create OAuth2 provider.
+        oauth2:OutboundOAuth2Provider oauth2Provider = new(salesforceConfig.clientConfig);
+        // Create bearer auth handler using created provider.
+        SalesforceBulkAuthHandler bearerHandler = new(oauth2Provider);
+
+        http:ClientSecureSocket? socketConfig = salesforceConfig?.secureSocketConfig;
+        
+        // Create salesforce http client.
+        if (socketConfig is http:ClientSecureSocket) {
+            self.httpClient = new(salesforceConfig.baseUrl, {
+                secureSocket: socketConfig,
+                auth: {
+                    authHandler: bearerHandler
+                }
+            });
+        } else {
+            self.httpClient = new(salesforceConfig.baseUrl, {
+                auth: {
+                    authHandler: bearerHandler
+                }
+            });
+        }
+    }
+
+    # Create a bulk job
+    #
+    # + operation - type of operation like insert, delete, etc
+    # + sobj - kind of sobject 
+    # + contentType - content type of the job 
+    # + return - returns job object or error
+    public remote function creatJob(string operation, string sobj, string contentType) returns @tainted ConnectorError|BulkJob {
```

Comment:
For the operation parameter, Is it possible to define an enum type with possible values and accept that type of values?

Reply:
done